include/onnxruntime/core/framework/ortdevice.h

```diff
@@ -71,8 +88,13 @@ struct OrtDevice {
       return device_type < other.device_type;
     if (memory_type != other.memory_type)
       return memory_type < other.memory_type;
+    if (device_id != other.device_id)
+      return device_id < other.device_id;
+
+    if (alignment && other.alignment)
+      return *alignment < *other.alignment;
```

Comment:
Do we need a default value if only one instance has alignment set? not perfect as you don't know the implicit default, but maybe better than assuming things are equal. 

related question would be whether we should provide the MLAS default alignment when creating the CPU allocator to make any implicit alignment value explicit so the comparison here is accurate. 

```suggestion
    auto align_to = alignment ? *alignment : 0;
    auto other_align_to = other.alignment ? *other.alignment : 0;
    if (align_to != other_align_to)
      return *alignment < *other.alignment;
```